ProductionModePlugin.js

```diff
@@ -191,66 +190,93 @@ ProductionModePlugin.prototype.apply = function(compiler) {
-            globalizeModuleIdsMap[aux] = module.id;
+
+            // some plugins, like HashedModuleIdsPlugin, may change module ids
+            // into strings.
+            var moduleId = module.id;
+            if (typeof moduleId === "string") {
+              moduleId = JSON.stringify(moduleId);
+            }
+
+            globalizeModuleIds.push(moduleId);
+            globalizeModuleIdsMap[aux] = moduleId;
           }
         });
       });
 
-      chunks.filter(function(chunk) {
-        return /globalize-compiled-data/.test(chunk.name);
-      }).forEach(function(chunk) {
-        var locale = chunk.name.replace("globalize-compiled-data-", "");
-        chunk.files.filter(ModuleFilenameHelpers.matchObject).forEach(function(file) {
-          var isFirst = true;
-          // Match either webpack 1.x or webpack 2.x
-          var source = compilation.assets[file].source().replace(/\n\/\*\*\*\/ (\(?function)\(module, exports(, __webpack_require__)?\) {[\s\S]*?(\n\/\*\*\*\/ })/g, function(garbage1, fnHead, garbage2, fnTail) {
-            var fnContent;
+      // rewrite the modules in the localized chunks:
+      //   - entry module will contain the compiled formatters and parsers
+      //   - non-entry modules will be rewritten to export globalize
+      this.chunks
+        .filter((chunk) => /globalize-compiled-data/.test(chunk.name))
+        .forEach((chunk) => {
+          // remove dead entry module for these reasons
+          //   - because the module has no dependencies, it won't be rendered
+          //     with __webpack_require__, making it difficult to modify its
+          //     source in a way that can import globalize
+          //
+          //   - it was a placeholder MultiModule that held no content, created
+          //     when we added a MultiEntryPlugin
+          //
+          //   - the true entry module should be globalize-compiled-data
+          //     module, which has been created as a NormalModule
+          chunk.removeModule(chunk.entryModule);
+          chunk.entryModule = chunk.modules.find((module) =>
+              module.context.endsWith(".tmp-globalize-webpack"));
 
-            // Define the initial module 0 as the whole formatters and parsers.
-            if (isFirst) {
-              isFirst = false;
+          const newModules = chunk.modules.map((module) => {
+            let fnContent;
+            if (module === chunk.entryModule) {
+              var locale = chunk.name.replace("globalize-compiled-data-", "");
               fnContent = globalizeCompilerHelper.compile(locale)
                 .replace("typeof define === \"function\" && define.amd", "false")
                 .replace(/require\("([^)]+)"\)/g, function(garbage, moduleName) {
                   return "__webpack_require__(" + globalizeModuleIdsMap[moduleName] + ")";
                 });
-
-            // Define all other individual globalize compiled data as a simple exports to Globalize.
```

Comment:
Should we leave this comment? Otherwise, the `else` code below would be pretty confusing...